functions.php

```diff
@@ -1,6 +1,177 @@
+	 */
+	public static function get_instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new WSU_Student_Financial_Services_Theme;
+			self::$instance->load_plugins();
+			self::$instance->setup_hooks();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Load "plugins" included with the theme.
+	 */
+	public function load_plugins() {
+		require_once( __DIR__ . '/includes/site-actions-widget.php' );
+	}
+
+	/**
+	 * Setup hooks to include.
+	 */
+	public function setup_hooks() {
+		add_action( 'init', array( $this, 'register_menu' ), 10 );
+		add_action( 'widgets_init', array( $this, 'register_sidebars' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+	}
+
+	/**
+	 * Register the additional menu used in the theme footer.
+	 */
+	public function register_menu() {
+		register_nav_menu( 'footer', 'Footer' );
+	}
+
+	/**
+	 * Register the sidebars and custom widgets used by the theme.
+	 */
+	public function register_sidebars() {
+		register_widget( 'Site_Actions_Widget' );
+
+		$footer_args = array(
+			'name' => 'Actions',
+			'id' => 'site-actions',
+			'description' => 'Displays the action links on the top of every page.',
+		);
+
+		register_sidebar( $footer_args );
+	}
+
+	/**
+	 * Enqueue the scripts used in the theme.
+	 */
+	public function enqueue_scripts() {
+		if ( is_front_page() ) {
+			wp_enqueue_script( 'sfs-scripts', get_stylesheet_directory_uri() . '/js/home.js', array( 'jquery' ), $this->script_version, true );
+		}
+
+		$post = get_post();
+
+		if ( isset( $post->post_content ) && strpos( $post->post_content, 'js-accordion' ) ) {
+			wp_enqueue_script( 'sfs-scripts', get_stylesheet_directory_uri() . '/js/accordion.js', array( 'jquery' ), $this->script_version, true );
+		}
+
+		if ( isset( $post->post_content ) && strpos( $post->post_content, 'dropdown' ) ) {
+			wp_enqueue_script( 'sfs-scripts', get_stylesheet_directory_uri() . '/js/table-toggle.js', array( 'jquery' ), $this->script_version, true );
+		}
+	}
+}
+
+add_action( 'after_setup_theme', 'WSU_Student_Financial_Services_Theme' );
 /**
- * Custom functionality required by your child theme can go here. Use this
- * to override any defaults provided by the Spine parent theme through
- * the provided actions and filters.
+ * Start things up.
+ *
+ * @return \WSU_Student_Financial_Services_Theme
  */
+function WSU_Student_Financial_Services_Theme() {
+	return WSU_Student_Financial_Services_Theme::get_instance();
+}
+
+/**
+ * Determine what should be displayed in the main header area.
+ *
+ * @return array List of elements for output in main header.
+ */
+function sfs_get_header_elements() {
+	$sfs_headers = array();
+
+	// Section title.
+	if ( is_page() ) {
+		// Retrieve the title of the top-level ancestor for pages.
+		$ancestor = get_page( array_pop( get_post_ancestors( get_the_ID() ) ) );
+		$sfs_headers['section_title'] = $ancestor->post_title;
+	} else if ( is_singular( 'post' ) || is_post_type_archive( 'post' ) ) {
```

Comment:
I'm not sure why yet, but it looks like `is_post_type_archive()` doesn't work for the `post` post type. If archive views for posts should have "Latest", then this needs to be `is_archive()`. (I think) 😄